libs/backend/src/cast_vote_records/import.ts

```diff
@@ -0,0 +1,268 @@
+      return;
+    }
+
+    // Only relevant for HMPBs
+    let castVoteRecordBallotSheetId: number | undefined;
+    if (castVoteRecord.BallotSheetId) {
+      const parseBallotSheetIdResult = safeParseNumber(
+        castVoteRecord.BallotSheetId
+      );
+      if (parseResult.isErr()) {
+        yield wrapError({ subType: 'invalid-ballot-sheet-id' });
+        return;
+      }
+      castVoteRecordBallotSheetId = parseBallotSheetIdResult.ok();
+    }
+
+    const castVoteRecordCurrentSnapshot = getCurrentSnapshot(castVoteRecord);
+    if (!castVoteRecordCurrentSnapshot) {
+      yield wrapError({ subType: 'no-current-snapshot' });
+      return;
+    }
+
+    const castVoteRecordWriteIns =
+      getWriteInsFromCastVoteRecord(castVoteRecord);
+    if (!castVoteRecordWriteIns.every(isCastVoteRecordWriteInValid)) {
+      yield wrapError({ subType: 'invalid-write-in-field' });
+      return;
+    }
+
+    let referencedFiles: ReferencedFiles | undefined;
+    if (castVoteRecord.BallotImage) {
+      if (
+        castVoteRecord.BallotImage.length !== 2 ||
+        !castVoteRecord.BallotImage[0]?.Location?.startsWith('file:') ||
+        !castVoteRecord.BallotImage[1]?.Location?.startsWith('file:')
+      ) {
+        yield wrapError({ subType: 'invalid-ballot-image-field' });
+        return;
+      }
+      const ballotImageLocations: [string, string] = [
+        castVoteRecord.BallotImage[0].Location.replace('file:', ''),
+        castVoteRecord.BallotImage[1].Location.replace('file:', ''),
+      ];
+
+      const imageFilePaths = ballotImageLocations.map((location) =>
+        path.join(castVoteRecordDirectoryPath, location)
+      ) as [string, string];
+      const layoutFilePaths = ballotImageLocations.map((location) =>
+        path.join(
+          castVoteRecordDirectoryPath,
+          `${path.parse(location).name}.layout.json`
+        )
+      ) as [string, string];
+
+      if (!imageFilePaths.every((filePath) => existsSync(filePath))) {
+        yield wrapError({ subType: 'image-file-not-found' });
+        return;
+      }
+      if (!layoutFilePaths.every((filePath) => existsSync(filePath))) {
+        yield wrapError({ subType: 'layout-file-not-found' });
+        return;
+      }
+
+      referencedFiles = { imageFilePaths, layoutFilePaths };
+    }
+
+    yield ok({
+      castVoteRecord,
+      castVoteRecordBallotSheetId,
+      castVoteRecordCurrentSnapshot,
+      castVoteRecordWriteIns,
+      referencedFiles,
+    });
+  }
+}
+
+/**
+ * Reads and parses a cast vote record export, authenticating the export in the process. The
+ * export's metadata file is parsed upfront whereas the cast vote records are parsed lazily.
+ *
+ * Basic validation is performed on the cast vote records. Referenced image files and layout files
+ * are not read/parsed, but their existence is validated such that consumers can safely access
+ * them.
+ */
+export async function readCastVoteRecordExport(
```

Comment:
My mental model for the separation of responsibilities between `libs/backend` and `apps/admin/backend` re import is that:

- `libs/backend` reads/parses data and performs validation internal to an export (i.e. it doesn't perform contextual validation like whether the fields in an export and the election definition correspond)
- `apps/admin/backend` actually persists data and performs contextual validation

This separation will allow us to reuse `libs/backend` import code in VxScan, when we build support for bootstrapping a VxScan with another VxScan's CVRs. The persistence logic and contextual validation for VxScan bootstrapping differ from the persistence logic and contextual validation for VxAdmin tabulation, so it's nice that those pieces are separated out.